apps/www/components/Audio/hooks/useMediaSession.tsx

```diff
@@ -0,0 +1,124 @@
+import { useCallback, useEffect, useRef } from 'react'
+import { CDN_FRONTEND_BASE_URL } from '../../../lib/constants'
+import { getImageCropURL } from '../AudioPlayer/ui/AudioCover'
+import { AudioQueueItem } from '../graphql/AudioQueueHooks'
+
+type PlayerState = {
+  duration: number
+  playbackRate: number
+  currentTime: number
+}
+
+type MediaSessionCallbacks = {
+  onPlay: () => void
+  onPause: () => void
+  onSeekForward: (seekOffset: number) => void
+  onSeekBackward: (seekOffset: number) => void
+  onSkipToNext: () => void
+  onStop: () => void
+  onRetrievePlayerState: () => PlayerState
+}
+
+type MediaSessionOptions = {
+  isPlaying: boolean
+  callbacks: MediaSessionCallbacks
+}
+
+export function useMediaSession(
+  playerItem: AudioQueueItem | null,
+  { isPlaying, callbacks }: MediaSessionOptions,
+) {
+  const callbackRefs = useRef(callbacks)
+
+  useEffect(() => {
+    callbackRefs.current = callbacks
+  }, [callbacks])
+
+  const updatePlayerState = useCallback(() => {
+    if ('mediaSession' in navigator) {
+      const playerState = callbackRefs.current.onRetrievePlayerState()
```

Comment:
Needs check if `current` is not undefined
```suggestion
      const playerState = callbackRefs.current?.onRetrievePlayerState()
```